crates/burn-fusion/src/stream/base.rs

```diff
@@ -4,7 +4,14 @@ use burn_tensor::repr::OperationDescription;
 
 /// A growing list of [tensor operation descriptions](OperationDescription).
 pub struct OperationQueue<R: FusionRuntime> {
+    /// List of operation descriptions. These contain the exact tensor IDs
+    /// and shapes so that kernels can be optimized correctly.
+    ///
+    /// The length of this list is the same as the length of the `operations` list.
     pub(crate) global: Vec<OperationDescription>,
+    /// List of operation descriptions. The tensor IDs and shapes are relative
+    /// because we don't need to know the exact values to determine which operations
+    /// can be fused.
```

Comment:
So
- relative IDs so that we can fuse/optimize kernels correctly without knowing their exact shape?
- absolute IDs is to execute kernels at runtime? The number of cubes, etc. might depend on the exact shape?

Reply:
Yes, but also the absolute IDs are always changing; it's a stream of tensor operations, not fixed tensor IDs. So, we will reuse the same optimization based on the comparison of the relative ID, since the same streaming pattern normally arises multiple times.